pkgs/applications/editors/amp/default.nix

```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchFromGitHub, rustPlatform, openssl, pkgconfig, python3, xorg, cmake, libgit2 }:
+
+rustPlatform.buildRustPackage {
+  name = "amp";
+  # The latest release (0.5.2) does not compile, so we use a git snapshot instead.
+  version = "2019.06.11";
+
+  src = fetchFromGitHub {
+    owner = "jmacdonald";
+    repo = "amp";
+    rev = "2c88e82a88ada8a5fd2620ef225192395a4533a2";
+    sha256 = "0ha1xiabq31s687gkrnszf3zc7b3sfdl79iyg5ygbc49mzvarp8c";
+  };
+
+  cargoSha256 = "1bvj2zg19ak4vi47vjkqlybz011kn5zq1j7zznr76zrryacw4lz1";
+
+  buildInputs = [ openssl pkgconfig python3 xorg.libxcb cmake libgit2 ];
```

Comment:
`cmake` and `pkgconfig` should be in `nativeBuildInputs` instead of `buildInputs`.

Reply:
I just looked at the repo. What are `cmake` and `pkgconfig` for? I don't think they're needed?

Reply:
pkgconfig:
```
error: failed to run custom build command for `openssl-sys v0.9.40`
process didn't exit successfully: `/build/source/target/release/build/openssl-sys-8aaaa48dbf1a2055/build-script-main` (exit code: 101)
--- stdout
cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_GNU_OPENSSL_LIB_DIR
cargo:rerun-if-env-changed=OPENSSL_LIB_DIR
cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_GNU_OPENSSL_INCLUDE_DIR
cargo:rerun-if-env-changed=OPENSSL_INCLUDE_DIR
cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_GNU_OPENSSL_DIR
cargo:rerun-if-env-changed=OPENSSL_DIR
run pkg_config fail: "Failed to run `\"pkg-config\" \"--libs\" \"--cflags\" \"openssl\"`: No such file or directory (os error 2)"
```

cmake:
```
error: failed to run custom build command for `onig_sys v68.0.1`
process didn't exit successfully: `/build/source/target/release/build/onig_sys-ada7042bd30242c4/build-script-build` (exit code: 101)
--- stdout
running: "cmake" "/build/amp-2019.06.11-vendor/onig_sys/oniguruma" "-DBUILD_SHARED_LIBS=OFF" "-DCMAKE_INSTALL_PREFIX=/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-e778458c74514984/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_C_COMPILER=/nix/store/1ap5d85630s3ksal6xgkjnbglmbng3kg-gcc-wrapper-7.4.0/bin/cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_CXX_COMPILER=/nix/store/1ap5d85630s3ksal6xgkjnbglmbng3kg-gcc-wrapper-7.4.0/bin/c++" "-DCMAKE_BUILD_TYPE=Release"
```

Reply:
@sbourdeauducq OK, looks good. Back to my comment on `nativeBuildInputs` then.